src/Explorer/Panes/UploadItemsPane/index.tsx

```diff
@@ -0,0 +1,220 @@
+    } else if (_totalFileSizeForFileList(files) > UPLOAD_FILE_SIZE_LIMIT) {
+      setFormError("Upload file size limit exceeded");
+      setFormErrorDetail("Total file upload size exceeds the 2 MB file size limit.");
+      logConsoleError("Could not upload items -- Total file upload size exceeds the 2 MB file size limit.");
+    }
+
+    const selectedCollection = explorer.findSelectedCollection();
+
+    setIsExecuting(true);
+
+    selectedCollection
+      ?.uploadFiles(files)
+      .then(
+        (uploadDetails: UploadDetails) => {
+          setUploadFileData(uploadDetails.data);
+          setSelectedFilesTitle("");
+          setFiles(undefined);
+        },
+        (error: Error) => {
+          const errorMessage = getErrorMessage(error);
+          setFormError(errorMessage);
+          setFormErrorDetail(errorMessage);
+        }
+      )
+      .finally(() => {
+        setIsExecuting(false);
+      });
+  };
+
+  const updateSelectedFiles = (event: ChangeEvent<HTMLInputElement>): void => {
+    const { files: newFiles } = event.target;
+
+    let newFileList = "";
+
+    if (!newFiles || newFiles.length === 0) {
+      return;
+    }
+
+    for (let i = 0; i < newFiles?.length; i++) {
+      newFileList += `"${newFiles.item(i).name}"`;
+    }
+    setFiles(newFiles);
+    setSelectedFilesTitle(newFileList);
+  };
+
+  const onImportLinkClick = (): void => {
+    fileRef?.current?.click();
+  };
+
+  const onImportLinkKeyPress = (event: KeyboardEvent<HTMLAnchorElement>): void => {
+    if (event.keyCode === Constants.KeyCodes.Enter || event.keyCode === Constants.KeyCodes.Space) {
+      onImportLinkClick();
+    }
+  };
+
+  const _totalFileSizeForFileList = (fileList: FileList): number => {
+    let totalFileSize = 0;
+    for (let i = 0; i < fileList?.length; i++) {
+      totalFileSize += fileList.item(i).size;
+    }
+    return totalFileSize;
+  };
+
+  const genericPaneProps: GenericRightPaneProps = {
+    container: explorer,
+    formError,
+    formErrorDetail,
+    id: "uploaditemspane",
+    isExecuting: isExecuting,
+    title: getTitle(),
+    submitButtonText: "Upload",
+    isSubmitButtonHidden: selectedFilesTitle === "",
+    onClose: closePanel,
+    onSubmit,
+  };
+  const columns: IColumn[] = [
+    {
+      key: "fileName",
+      name: "FILE NAME",
+      fieldName: "fileName",
+      minWidth: 140,
+      maxWidth: 140,
+    },
+    {
+      key: "status",
+      name: "STATUS",
+      fieldName: "numSucceeded",
+      minWidth: 140,
+      maxWidth: 140,
+      isRowHeader: true,
+      isResizable: true,
+      data: "string",
+      isPadded: true,
+    },
+  ];
+
+  const _renderItemColumn = (item: IUploadFileData, index: number, column: IColumn) => {
+    switch (column.key) {
+      case "status":
+        return <span>{item?.numSucceeded + " items created, " + item?.numFailed + " errors"}</span>;
```

Comment:
if `item` is `undefined`, this would probably not render well. Consider check for `undefined` earlier and render something different and here remove the optional operator `?`.

Reply:
remove optional operator ? . i thing not need to check item is null.